runtime/interpreter/value.go

```diff
@@ -449,29 +515,30 @@ func (v *StringValue) RecursiveString(_ SeenReferences) string {
 	return v.String()
 }
 
-func (v *StringValue) KeyString() string {
-	return v.Str
-}
-
-func (v *StringValue) Equal(other Value, _ *Interpreter, _ bool) bool {
+func (v *StringValue) Equal(_ *Interpreter, _ func() LocationRange, other Value) bool {
 	otherString, ok := other.(*StringValue)
 	if !ok {
 		return false
 	}
 	return v.NormalForm() == otherString.NormalForm()
 }
 
+func (v *StringValue) HashInput(_ *Interpreter, _ func() LocationRange, _ []byte) []byte {
+	return append(
+		[]byte{HashInputTypeString},
+		v.Str...,
```

Comment:
@turbolent Optimize: use scratch

Reply:
@turbolent Limit result length to 1024. Same for other variable-sized hash input results (e.g. `IntValue`)